bot/cogs/tags.py

```diff
@@ -0,0 +1,266 @@
+
+from bot.constants import ADMIN_ROLE, MODERATOR_ROLE, OWNER_ROLE
+from bot.constants import SITE_API_TAGS_URL, TAG_COOLDOWN
+from bot.decorators import with_role
+from bot.utils import paginate
+
+
+class Tags:
+    """
+    Save new tags and fetch existing tags.
+    """
+
+    def __init__(self, bot: AutoShardedBot):
+        self.bot = bot
+        self.api_key = os.environ.get("BOT_API_KEY")
+        self.tag_cooldowns = {}
+
+    async def get_tag_data(self, tag_name: Union[str, None] = None) -> dict:
+        """
+        Retrieve the tag_data from our API
+
+        :param tag_name: the tag to retrieve
+        :return:
+        if tag_name was provided, returns a dict with tag data.
+        if not, returns a list of dicts with all tag data.
+
+        """
+        headers = {"X-API-KEY": self.api_key}
+        params = {}
+
+        if tag_name:
+            params['tag_name'] = tag_name
+
+        async with ClientSession() as session:
+            response = await session.get(SITE_API_TAGS_URL, headers=headers, params=params)
+            tag_data = await response.json()
+
+        return tag_data
+
+    async def post_tag_data(self, tag_name: str, tag_content: str) -> dict:
+        """
+        Send some tag_data to our API to have it saved in the database.
+
+        :param tag_name: The name of the tag to create or edit.
+        :param tag_content: The content of the tag.
+        :return: json response from the API in the following format:
+        {
+            'success': bool
+        }
+        """
+
+        headers = {"X-API-KEY": self.api_key}
+        params = {
+            'tag_name': tag_name,
+            'tag_content': tag_content
+        }
+
+        async with ClientSession() as session:
+            response = await session.post(SITE_API_TAGS_URL, headers=headers, json=params)
+            tag_data = await response.json()
+
+        return tag_data
+
+    async def delete_tag_data(self, tag_name: str) -> dict:
+        """
+        Delete a tag using our API.
+
+        :param tag_name: The name of the tag to delete.
+        :return: json response from the API in the following format:
+        {
+            'success': bool
+        }
+        """
+
+        headers = {"X-API-KEY": self.api_key}
+        params = {}
+
+        if tag_name:
+            params['tag_name'] = tag_name
+
+        async with ClientSession() as session:
+            response = await session.delete(SITE_API_TAGS_URL, headers=headers, json=params)
+            tag_data = await response.json()
+
+        return tag_data
+
+    @command(name="tags()", aliases=["tags"], hidden=True)
+    async def info(self, ctx: Context):
+        """
+        Show available methods for this class.
+
+        :param ctx: Discord message context
+        """
+
+        return await ctx.invoke(self.bot.get_command("help"), "Tags")
+
+    @command(name="tags.get()", aliases=["tags.get", "tags.show()", "tags.show"])
```

Comment:
So far, I've been adding command aliases that make sense without the namespace, but that's entirely up to you I think. Like, ClickUp has `open_task` for example.